pkg/catalogv2/git/download.go

```diff
@@ -1,124 +1,101 @@
-	git, err := gitForRepo(secret, namespace, name, gitURL, insecureSkipTLS, caBundle)
+// Update will check if repository exists, if exists will check for latest commit and update to it.
+// If the repository does not exist will try cloning again.
+func (r *Repository) Update(branch string) (string, error) {
+	err := r.cloneOrOpen(branch)
 	if err != nil {
-		return err
+		return "", fmt.Errorf("Update failure: %w", err)
 	}
 
-	return git.Ensure(branch)
-}
-
-func isBundled(git *git) bool {
-	return strings.HasPrefix(git.Directory, staticDir) || strings.HasPrefix(git.Directory, localDir)
-}
-
-func gitForRepo(secret *corev1.Secret, namespace, name, gitURL string, insecureSkipTLS bool, caBundle []byte) (*git, error) {
-	isGitSSH, err := isGitSSH(gitURL)
+	err = r.hardReset("HEAD")
 	if err != nil {
-		return nil, fmt.Errorf("failed to verify the type of URL %s: %w", gitURL, err)
+		return "", fmt.Errorf("Update failure: %w", err)
 	}
-	if !isGitSSH {
-		u, err := url.Parse(gitURL)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse URL %s: %w", gitURL, err)
-		}
-		if u.Scheme != "http" && u.Scheme != "https" {
-			return nil, fmt.Errorf("invalid git URL scheme %s, only http(s) and git supported", u.Scheme)
-		}
+
+	commit, err := r.getCurrentCommit()
+	if err != nil {
+		return commit.String(), fmt.Errorf("Update failure: %w", err)
 	}
-	dir := gitDir(namespace, name, gitURL)
-	headers := map[string]string{}
-	if settings.InstallUUID.Get() != "" {
-		headers["X-Install-Uuid"] = settings.InstallUUID.Get()
+
+	lastCommit, err := r.getLastCommitHash(branch, commit)
+	if err != nil {
+		return commit.String(), fmt.Errorf("Update failure: %w", err)
 	}
-	// convert caBundle to PEM format because git requires correct line breaks, header and footer.
-	if len(caBundle) > 0 {
-		caBundle = convertDERToPEM(caBundle)
-		insecureSkipTLS = false
+	if lastCommit == commit {
+		return commit.String(), nil
 	}
-	return newGit(dir, gitURL, &Options{
-		Credential:        secret,
-		Headers:           headers,
-		InsecureTLSVerify: insecureSkipTLS,
-		CABundle:          caBundle,
-	})
-}
 
-func isGitSSH(gitURL string) (bool, error) {
-	// Matches URLs with the format [anything]@[anything]:[anything]
-	return regexp.MatchString("(.+)@(.+):(.+)", gitURL)
-}
-
-func hash(gitURL string) string {
-	b := sha256.Sum256([]byte(gitURL))
-	return hex.EncodeToString(b[:])
-}
+	err = r.fetchAndReset(branch)
+	if err != nil {
+		return commit.String(), fmt.Errorf("Update failure: %w", err)
+	}
 
-// convertDERToPEM converts a src DER certificate into PEM with line breaks, header, and footer.
-func convertDERToPEM(src []byte) []byte {
-	return pem.EncodeToMemory(&pem.Block{
-		Type:    "CERTIFICATE",
-		Headers: map[string]string{},
-		Bytes:   src,
-	})
+	lastCommitRef, err := r.getCurrentCommit()
+	if err != nil {
+		return lastCommit.String(), fmt.Errorf("Update failure: %w", err)
```

Comment:
Shouldn't this be lastCommitRef.String() ?

Reply:
yes, fixing

Reply:
fixed, please resolve